### PR TITLE
Added DPG methods to `AppConfiguration` client

### PR DESCRIPTION
**Pull request by DivineNwaokocha**
Integrated DPG methods into the `AppConfiguration` client. Now, the existing `AppConfiguration` methods call the generated methods. Changes were also made to `ConfigurationSetting` and `ConfigurationClientOptions` to facilitate the DPG integration.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj [hunk 1]
```diff
@@ -8,31 +8,35 @@
...
     <Compile Include="$(AzureCoreSharedSources)ConditionalRequestOptionsExtensions.cs" LinkBase="Shared" />
-    <Compile Include="$(AzureCoreSharedSources)DiagnosticScope.cs" LinkBase="Shared" />
+  <!--  <Compile Include="$(AzureCoreSharedSources)DiagnosticScope.cs" LinkBase="Shared" /> -->
     <Compile Include="$(AzureCoreSharedSources)HashCodeBuilder.cs" LinkBase="Shared" />
     <Compile Include="$(AzureCoreSharedSources)NoBodyResponseOfT.cs" LinkBase="Shared" />
     <Compile Include="$(AzureCoreSharedSources)PageResponseEnumerator.cs" LinkBase="Shared" />
-    <Compile Include="$(AzureCoreSharedSources)DiagnosticScopeFactory.cs" LinkBase="Shared" />
-    <Compile Include="$(AzureCoreSharedSources)TaskExtensions.cs" LinkBase="Shared" />
+  <!--  <Compile Include="$(AzureCoreSharedSources)DiagnosticScopeFactory.cs" LinkBase="Shared" />
+    <Compile Include="$(AzureCoreSharedSources)TaskExtensions.cs" LinkBase="Shared" /> -->
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj:39]: nit: remove commented types.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 2]
```diff
@@ -159,6 +160,24 @@ public virtual async Task<Response<ConfigurationSetting>> AddConfigurationSettin
...
+        private static RequestContent CreateContent(ConfigurationSetting setting)
+        {
+            Argument.AssertNotNull(setting, nameof(setting));
+            Argument.AssertNotNullOrEmpty(setting.Key, $"{nameof(setting)}.{nameof(setting.Key)}");
+
+            ReadOnlyMemory<byte> serializedSetting = ConfigurationServiceSerializer.SerializeRequestBody(setting);
+            RequestContent content = RequestContent.Create(serializedSetting);
+
+            return content;
+        }
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:172]: Can we move this method to `ConfigurationSetting` to make Client class cleaner? you can also create a partial class and add `ToRequestContent` method, and then you can call it from any method in the client like
 ```csharp
RequestContent content = ConfigurationSetting.ToRequestContent(setting)
```

You can refer this for more details : https://github.com/Azure/azure-sdk-for-net/blob/main/doc/DataPlaneCodeGeneration/DeveloperDrivenEvolution.md#create-the-models

Here is an example: https://github.com/ShivangiReja/azure-sdk-for-net/blob/MetricsAdvisor-Experiment/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeedDetail.Serialization.cs#L487-L498

Here is the usecase: https://github.com/ShivangiReja/azure-sdk-for-net/blob/MetricsAdvisor-Experiment/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs#L361

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 3]
```diff
@@ -159,6 +160,24 @@ public virtual async Task<Response<ConfigurationSetting>> AddConfigurationSettin
...
+
+        private static RequestContext CreateContext(CancellationToken cancellationToken)
+        {
+            RequestContext context = new RequestContext();
+            context.CancellationToken = cancellationToken;
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:177]: nit: you can simplify it to something like:
```csharp
            RequestContext context = new RequestContext()
            {
                CancellationToken = cancellationToken,
            };
```

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 4]
```diff
@@ -405,23 +427,28 @@ public virtual Response DeleteConfigurationSetting(ConfigurationSetting setting,
...
             scope.Start();
 
             try
             {
-                using Request request = CreateDeleteRequest(key, label, requestOptions);
-                Response response = await _pipeline.SendRequestAsync(request, cancellationToken).ConfigureAwait(false);
+                var eTag = requestOptions?.IfMatch;
+                if (requestOptions != null && requestOptions.IfMatch.HasValue)
+                {
+                     eTag = new ETag($"\"{requestOptions?.IfMatch.Value.ToString()}\"");
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:439]: Looking at [ConditionalRequestOptionsExtensions.ApplyHeaders](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/ConditionalRequestOptionsExtensions.cs#L14-L15), we don't need `$"\"{requestOptions?.IfMatch.Value.ToString()}\""`  when `options.IfMatch.Value` is equals to `ETag.All`. Can we apply [this](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/ConditionalRequestOptionsExtensions.cs#L14-L15) similar condition here?

Also, could you please create an issue in [c# generator](https://github.com/Azure/autorest.csharp/issues/new/choose) about [this](https://github.com/DivineNwaokocha/azure-sdk-for-net/blob/43a9c2ff42fa0b84b01a30831b15d8ba8fcc2a19/sdk/appconfiguration/Azure.Data.AppConfiguration/src/Generated/ConfigurationClient.cs#L2113) should call [ConditionalRequestOptionsExtensions.ApplyHeaders](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/ConditionalRequestOptionsExtensions.cs#L10) method when applying `MatchConditions` header, so we don't have to add this here?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/autorest.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+``` yaml
+input-file:
+- https://github.com/Azure/azure-rest-api-specs/blob/e01d8afe9be7633ed36db014af16d47fec01f737/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/appconfiguration.json
+namespace: Azure.Data.AppConfiguration
+title: ConfigurationClient
+public-clients: true
+directive:
+  from: swagger-document
+  where: $.parameters.Endpoint
+  transform: $.format = "url"
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/autorest.md:16]: Can we add a seperate directive for `Endpoint` similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/anomalydetector/Azure.AI.AnomalyDetector/src/autorest.md#make-endpoint-type-as-uri) with a comment to make it more clear?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj [hunk 6]
```diff
@@ -8,31 +8,35 @@
     <ApiCompatVersion>1.2.0</ApiCompatVersion>
     <PackageTags>Microsoft Azure Application Configuration;Data;AppConfig;$(PackageCommonTags)</PackageTags>
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
-    <NoWarn>$(NoWarn);3021</NoWarn>
+    <NoWarn>$(NoWarn);3021;AZC0007;AZC0006</NoWarn>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <DefineConstants>$(DefineConstants);EXPERIMENTAL</DefineConstants>
```
Comment (jsquire) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj:13]: Interesting...  what causes us to need the `Experimental` constant?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 7]
```diff
@@ -132,24 +129,28 @@ public virtual Response<ConfigurationSetting> AddConfigurationSetting(string key
...
+            using DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(ConfigurationClient)}.{nameof(AddConfigurationSetting)}");
             scope.AddAttribute("key", setting?.Key);
             scope.Start();
 
             try
             {
-                using Request request = CreateAddRequest(setting);
-                Response response = await _pipeline.SendRequestAsync(request, cancellationToken).ConfigureAwait(false);
+                RequestContext context = CreateContext(cancellationToken);
+                RequestContent content = CreateContent(setting);
```
Comment (jsquire) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:139]: `RequestContent` is disposable.   We'll want to be sure that we do so.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 8]
```diff
@@ -132,24 +129,28 @@ public virtual Response<ConfigurationSetting> AddConfigurationSetting(string key
...
+                ContentType contentType = new ContentType(HttpHeader.Common.JsonContentType.Value.ToString());
+                MatchConditions requestOptions = new MatchConditions { IfNoneMatch = ETag.All };
+
+                Response response = await PutKeyValueAsync(setting.Key, content, contentType, setting.Label, requestOptions, context).ConfigureAwait(false);
```
Comment (jsquire) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:143]: `Response` is disposable; we should consider adding a note to the `<returns>` area of the docs to let callers know they hold responsibility for disposing it.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj (hunk 6 above)
Comment (DivineNwaokocha) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj:13]: It was initially copied from the other DPG libraries (e.g. https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/Azure.IoT.DeviceUpdate.csproj#L12), but I have just updated it and after removing the experimental dependency, the code runs fine, thanks!

File: common/Perf/Azure.Test.Perf/NumberFormatter.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Test.Perf
```
Comment (jsquire) [inline common/Perf/Azure.Test.Perf/NumberFormatter.cs:6]: Definitely shouldn't be picking these up in the PR.   You'll most likely need to rebase to `main` in order to ensure that the only files here are those you're intending to change.

File: sdk/appconfiguration/package-lock.json [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "appconfiguration",
+  "lockfileVersion": 2,
+  "requires": true,
+  "packages": {}
+}
```
Comment (ShivangiReja) [inline sdk/appconfiguration/package-lock.json:6]: Do need this file?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs (hunk 7 above)
Comment (DivineNwaokocha) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]: Thanks for the catch! I've opted to include the using statement which will call the dispose method for the object.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs (hunk 8 above)
Comment (DivineNwaokocha) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]: Is this still necessary if the using statement handles the disposal?

File: sdk/appconfiguration/package-lock.json (hunk 10 above)
Comment (DivineNwaokocha) [reply sdk/appconfiguration/package-lock.json]: No we do not, thanks!

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj [hunk 11]
```diff
@@ -8,31 +8,26 @@
...
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
-    <NoWarn>$(NoWarn);3021</NoWarn>
+    <NoWarn>$(NoWarn);3021;AZC0007;AZC0006</NoWarn>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <IncludeOperationsSharedSource>true</IncludeOperationsSharedSource> 
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="Azure.Core" />
+    <PackageReference Include="Azure.Core.Experimental" />
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj:18]: Do we need `Azure.Core.Experimental` package reference?

Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj:18]: Do we need `Azure.Core.Experimental` package reference? if not, can we remove it?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs (hunk 2 above)
Comment (DivineNwaokocha) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]: Great idea, I moved both `CreateContent` and `CreateContext` to `ConfigurationSetting`

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj (hunk 11 above)
Comment (DivineNwaokocha) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj]: No It's not needed and it's been removed, thanks!

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs (hunk 2 above)
Comment (ShivangiReja) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]: We don't need to move `CreateContext` to `ConfigurationSetting`. 

Also, can we rename `CreateContent` to `ToRequestContent` as per [this](https://github.com/Azure/azure-sdk-for-net/blob/main/doc/DataPlaneCodeGeneration/DeveloperDrivenEvolution.md#create-the-models) doc?

Comment (ShivangiReja) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]: @DivineNwaokocha  can we rename `CreateContent` to `ToRequestContent` as per [this](https://github.com/Azure/azure-sdk-for-net/blob/main/doc/DataPlaneCodeGeneration/DeveloperDrivenEvolution.md#create-the-models) doc?

Comment (DivineNwaokocha) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]: Hi, I renamed it, I'm just working through an error before I push

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs [hunk 12]
```diff
@@ -77,6 +78,14 @@ internal virtual void SetValue(string value)
             _value = value;
         }
 
+        internal static RequestContent CreateContent(ConfigurationSetting setting)
+        {
+            ReadOnlyMemory<byte> serializedSetting = ConfigurationServiceSerializer.SerializeRequestBody(setting);
+            RequestContent content = RequestContent.Create(serializedSetting);
+
+            return content;
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs:86]: nit: you can simplify this
```csharp
return RequestContent.Create(serializedSetting);
````

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/Generated/ConfigurationClient.cs [hunk 13]
```diff
@@ -0,0 +1,2437 @@
...
+                request.Headers.Add("Sync-Token", _syncToken);
+            }
+            if (acceptDatetime != null)
+            {
+                request.Headers.Add("Accept-Datetime", acceptDatetime);
+            }
+            request.Headers.Add("Accept", "application/vnd.microsoft.appconfig.kv+json, application/json, application/problem+json");
+            if (matchConditions != null)
+            {
+                request.Headers.Add(matchConditions);
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/Generated/ConfigurationClient.cs:2057]: You need to regenerate the code again.

Based on https://github.com/Azure/autorest.csharp/pull/2561/files changes, it should generate something like [this](https://github.com/archerzz/autorest.csharp/blob/2f80142d8e85f8d7c89ca803e92ec858e4b5f6a5/src/assets/Generator.Shared/RequestHeaderExtensions.cs#L69).

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/Generated/ConfigurationClient.cs [hunk 14]
```diff
@@ -0,0 +1,2437 @@
...
+            uri.AppendQuery("api-version", _apiVersion, true);
+            request.Uri = uri;
+            if (_syncToken != null)
+            {
+                request.Headers.Add("Sync-Token", _syncToken);
+            }
+            request.Headers.Add("Accept", "application/vnd.microsoft.appconfig.kv+json, application/json, application/problem+json");
+            if (ifMatch != null)
+            {
+                request.Headers.Add("If-Match", ifMatch.Value);
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/Generated/ConfigurationClient.cs:2113]: You need to regenerate the code again.

Based on https://github.com/Azure/autorest.csharp/pull/2561/files changes, it should generate something like [this](https://github.com/archerzz/autorest.csharp/blob/2f80142d8e85f8d7c89ca803e92ec858e4b5f6a5/src/assets/Generator.Shared/RequestHeaderExtensions.cs#L69).

File: sdk/appconfiguration/Azure.Data.AppConfiguration/CHANGELOG.md [hunk 15]
```diff
@@ -3,12 +3,7 @@
 ## 1.3.0-beta.1 (Unreleased)
 
 ### Features Added
-
-### Breaking Changes
-
-### Bugs Fixed
-
-### Other Changes
+- Added DPG methods to `ConfigurationClient` based on this swagger definition : https://github.com/Azure/azure-rest-api-specs/blob/e01d8afe9be7633ed36db014af16d47fec01f737/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/appconfiguration.json
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/CHANGELOG.md:6]:
```suggestion
- Added DPG methods to `ConfigurationClient` based on [this](https://github.com/Azure/azure-rest-api-specs/blob/e01d8afe9be7633ed36db014af16d47fec01f737/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/appconfiguration.json) swagger definition.
```

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj [hunk 16]
```diff
@@ -8,7 +8,9 @@
     <ApiCompatVersion>1.2.0</ApiCompatVersion>
     <PackageTags>Microsoft Azure Application Configuration;Data;AppConfig;$(PackageCommonTags)</PackageTags>
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
-    <NoWarn>$(NoWarn);3021</NoWarn>
+    <NoWarn>$(NoWarn);3021;AZC0007;AZC0006</NoWarn>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj:12]: Why do we need `AllowUnsafeBlocks` flag as true?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 17]
```diff
@@ -50,11 +47,11 @@ public ConfigurationClient(string connectionString, ConfigurationClientOptions o
                 throw new ArgumentNullException(nameof(options));
 
             ParseConnectionString(connectionString, out _endpoint, out var credential, out var secret);
-
+            _apiVersion = options.Version;
             _syncTokenPolicy = new SyncTokenPolicy();
             _pipeline = CreatePipeline(options, new AuthenticationPolicy(credential, secret), _syncTokenPolicy);
 
-            _clientDiagnostics = new ClientDiagnostics(options);
+            ClientDiagnostics = new ClientDiagnostics(options);
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:54]: Pass `suppressNestedClientActivities` as true in `new ClientDiagnostics(options, true);` to disable nested scopes.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 18]
```diff
@@ -80,16 +77,42 @@ public ConfigurationClient(Uri endpoint, TokenCredential credential, Configurati
 
             _endpoint = endpoint;
             _syncTokenPolicy = new SyncTokenPolicy();
+            _syncToken = _syncTokenPolicy.ToString();
             _pipeline = CreatePipeline(options, new BearerTokenAuthenticationPolicy(credential, GetDefaultScope(endpoint)), _syncTokenPolicy);
+            _apiVersion = options.Version;
+
+            ClientDiagnostics = new ClientDiagnostics(options);
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:84]: Pass `suppressNestedClientActivities` as true in `new ClientDiagnostics(options, true);` to disable nested scopes.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 19]
```diff
@@ -693,193 +712,12 @@ public virtual AsyncPageable<ConfigurationSetting> GetRevisionsAsync(SettingSele
...
-            return PageResponseEnumerator.CreateEnumerable(nextLink => GetRevisionsPage(selector, nextLink, cancellationToken));
-        }
-
-        private Request CreateGetRequest(string key, string label, DateTimeOffset? acceptDateTime, MatchConditions requestOptions)
-        {
-            Argument.AssertNotNullOrEmpty(key, nameof(key));
-
-            Request request = _pipeline.CreateRequest();
-            request.Method = RequestMethod.Get;
-            BuildUriForKvRoute(request.Uri, key, label);
```
Comment (ShivangiReja) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs:705]: Can you please clean up all the unused methods?

For example: Now we don't need [these](https://github.com/Azure/azure-sdk-for-net/blob/8ad3e7480fa1ee9695dc114354d57b5af1f07594/sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient_private.cs#L61-L141) methods, so we can remove it.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj (hunk 16 above)
Comment (DivineNwaokocha) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/Azure.Data.AppConfiguration.csproj:12]: This was copied over from another sdk in an attempt to fix an earlier error, It's not necessary.

File: sdk/identity/Azure.Identity/src/Azure.Identity.csproj [hunk 20]
```diff
@@ -8,7 +8,6 @@
     <PackageTags>Microsoft Azure Identity;$(PackageCommonTags)</PackageTags>
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
     <NoWarn>$(NoWarn);3021;AZC0011</NoWarn>
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (KrzysztofCwalina) [inline sdk/identity/Azure.Identity/src/Azure.Identity.csproj:11]: Why is this needed?

Comment (DivineNwaokocha) [reply sdk/identity/Azure.Identity/src/Azure.Identity.csproj]: It's not! It was removed unnecessarily and has been undone.

**Review by ShivangiReja — APPROVED** [review]
